### PR TITLE
Fixes #13542 - Add in the translation scaffolding for foreman-tasks

### PR DESCRIPTION
**Pull request by bkearney**
After this is merged. It will need to be added to transifex @claer, @stbenjam, @orrabin

### CODE REVIEW

File: locale/action_names.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Autogenerated!
+_("Delete architecture")
+_("Import puppetclasses")
```
Comment (domcleal) [inline locale/action_names.rb]: "Puppet classes"

File: locale/foreman_tasks.pot [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:87
+msgid "Resume all paused error tasks"
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:88
+msgid "Resume tasks matching search string"
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:89
+msgid "Resume specific tasks by id"
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: `ID`

File: locale/foreman_tasks.pot [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:126
+msgid "Page number, starting at 1"
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:127
+msgid "Number of results per page to return"
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:128
+msgid "Sort field and order, eg. 'name DESC'"
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: `e.g.`

File: locale/foreman_tasks.pot [hunk 4]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:164
+msgid "Send data to the task from external executor (such as smart_proxy_dynflow)"
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:166
+msgid "UUID of the task"
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:167
+msgid "The id of the step inside the execution plan to send the event to "
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: `ID`, trailing whitespace

File: locale/foreman_tasks.pot [hunk 5]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:205
+msgid "Resource search_params requires resource_type and resource_id to be specified"
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:212
+msgid "Task search_params requires task_id to be specified"
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/api/tasks_controller.rb:216
+msgid "Search_Params %s not supported"
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: Strange capitalisation of the first bit?

File: locale/foreman_tasks.pot [hunk 6]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/controllers/foreman_tasks/tasks_controller.rb:24
+msgid "Trying to cancel step %s"
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/tasks_controller.rb:32
+msgid "Trying to cancel the task"
+msgstr ""
+
+#: ../app/controllers/foreman_tasks/tasks_controller.rb:34
+msgid "The task is not cancellable at the moment."
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: cannot be cancelled?

File: locale/foreman_tasks.pot [hunk 7]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/views/foreman_tasks/tasks/_details.html.erb:50
+#: ../app/views/foreman_tasks/tasks/_details.html.erb:75
+#: ../app/views/foreman_tasks/tasks/_running_steps.html.erb:12
+msgid "Cancel"
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:28
+#: ../app/views/foreman_tasks/recurring_logics/index.html.erb:22
+#: ../app/views/foreman_tasks/task_groups/recurring_logic_task_groups/_recurring_logic_task_group.html.erb:17
+msgid "-"
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: Needs more context or to be removed.

File: locale/foreman_tasks.pot [hunk 8]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:97
+msgid "Explanation"
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:75
+msgid "Repeats"
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:90
+msgid "is minute (range: 0-59)"
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: A little more context may help, even a TRANSLATORS comment.

File: locale/foreman_tasks.pot [hunk 9]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:93
+msgid "is month (range: 1-12)"
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:94
+msgid "is day of week (range: 0-6)"
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:98
+msgid "\"Cron line format 'a b c d e', where:<br><ol type=\\\"a\\\">#{options}</ol>\""
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: Leading/trailing quotes, HTML and interpolation should be removed.

File: locale/foreman_tasks.pot [hunk 10]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:139
+msgid "Ends at"
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/tasks_helper.rb:18
+msgid "Unlimited"
+msgstr ""
+
+#: ../app/lib/actions/action_with_sub_plans.rb:15
+msgid "%{total} tasks, %{success} success, %{failed} fail"
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: missing plural forms?

File: locale/foreman_tasks.pot [hunk 11]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:134
+msgid "Ends"
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:135
+msgid "Never"
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:136
+msgid "On"
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: Context may help.

File: locale/foreman_tasks.pot [hunk 12]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:122
+msgid "Sat"
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:123
+msgid "Sun"
+msgstr ""
+
+#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:131
+msgid "At"
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: Missing context?

File: locale/foreman_tasks.pot [hunk 13]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/lib/actions/foreman/architecture/update.rb:11
+msgid "Update architecture"
+msgstr ""
+
+#: ../app/lib/actions/foreman/host/import_facts.rb:35
+msgid "Import facts"
+msgstr ""
+
+#: ../app/lib/actions/foreman/puppetclass/import.rb:18
+msgid "Import puppetclasses"
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: "Puppet classes"

File: locale/foreman_tasks.pot [hunk 14]
```diff
@@ -0,0 +1,725 @@
...
+#: ../app/lib/actions/helpers/humanizer.rb:187
+msgid "system"
+msgstr ""
+
+#: ../app/lib/actions/helpers/humanizer.rb:203
+msgid "organization"
+msgstr ""
+
+#: ../app/lib/actions/proxy_action.rb:60
+msgid "\"The smart proxy task '#{output[:proxy_task_id]}' failed.\""
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: Quotes and interpolation need to be removed.

File: locale/foreman_tasks.pot [hunk 15]
```diff
@@ -0,0 +1,725 @@
...
+msgid "%{value} is not allowed input type"
+msgstr ""
+
+#: ../app/models/foreman_tasks/triggering.rb:37
+#: ../app/models/foreman_tasks/triggering.rb:39
+msgid "%{value} is wrong format"
+msgstr ""
+
+#: ../app/models/foreman_tasks/triggering.rb:99
+msgid "\"#{cronline} is not valid format of cronline\""
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: Quotes and interpolation need to be removed.  "cronline" might need a space.

File: locale/foreman_tasks.pot [hunk 16]
```diff
@@ -0,0 +1,725 @@
...
+msgstr ""
+
+#: ../app/views/foreman_tasks/recurring_logics/index.html.erb:1
+#: ../lib/foreman_tasks/engine.rb:48
+msgid "Recurring logics"
+msgstr ""
+
+#: ../app/views/foreman_tasks/recurring_logics/index.html.erb:6
+#: ../app/views/foreman_tasks/task_groups/recurring_logic_task_groups/_recurring_logic_task_group.html.erb:8
+msgid "Cronline"
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: ditto above, may need a space?

File: locale/foreman_tasks.pot [hunk 17]
```diff
@@ -0,0 +1,725 @@
...
+msgid "External Id"
+msgstr ""
+
+#: ../app/views/foreman_tasks/tasks/_running_steps.html.erb:5
+msgid "No running steps"
+msgstr ""
+
+#:
+#: ../app/views/foreman_tasks/tasks/dashboard/_latest_tasks_in_error_warning.html.erb:1
+msgid "Latest Warning/Error Tasks "
```
Comment (domcleal) [inline locale/foreman_tasks.pot]: Trailing whitespace.

**Comment by bkearney** [comment]
@domcleal I addressed all your comments except 1 (see below). I did a force push with the latest strings.   The one comment I did not addres was

> +
> +#: ../app/helpers/foreman_tasks/foreman_tasks_helper.rb:98
> +msgid "\"Cron line format 'a b c d e', where:<br><ol type=\\\"a\\\">#{options}</ol>\""

Leading/trailing quotes, HTML and interpolation should be removed.

I removed the interpolation, but it seems that using the popover command you need to pass in html snippets.

File: app/helpers/foreman_tasks/foreman_tasks_helper.rb [hunk 18]
```diff
@@ -87,6 +87,7 @@ def recurring_mode_fieldset(f, triggering)
 
     def cronline_fieldset(f, triggering)
       options = [
+        # TRANSLATORS: these translations are ferring to an option which is a time interval
```
Comment (domcleal) [inline app/helpers/foreman_tasks/foreman_tasks_helper.rb]: referring, and this comment won't work across multiple lines I don't think?

File: app/helpers/foreman_tasks/foreman_tasks_helper.rb [hunk 19]
```diff
@@ -95,7 +96,7 @@ def cronline_fieldset(f, triggering)
       ].map { |opt| content_tag(:li, opt) }.join
       help = content_tag(:span, nil, :class => 'help-inline') do
         popover(_('Explanation'),
-                _("Cron line format 'a b c d e', where:<br><ol type=\"a\">#{options}</ol>"))
+                _("Cron line format 'a b c d e', where:<br><ol type=\"a\">%s</ol>") % (options))
```
Comment (domcleal) [inline app/helpers/foreman_tasks/foreman_tasks_helper.rb]: Pass the HTML in through interpolation rather than in the source string.

**Comment by bkearney** [comment]
@domcleal addressed both comments. I did need to replicate the TRANSLATORS comment because it did not get applied to subsequent translations.

File: locale/action_names.rb [hunk 20]
```diff
@@ -0,0 +1,54 @@
+# Autogenerated!
+_("Update package")
```
Comment (domcleal) [inline locale/action_names.rb]: These look like they're from a plugin you have loaded using foreman-tasks, not foreman-tasks itself.  Wrong project?

**Comment by bkearney** [comment]
@domclel updated. I was not aware that string extraction required that only a single plugin be active.

**Comment by domcleal** [comment]
It doesn't usually, it's probably the gettext extensions within foreman-tasks.

The POT looks OK for translation to me now.

**Comment by iNecas** [comment]
Thanks @bkearney and @domcleal

**Pull request merged**